apps/email/js/html_cache_restore.js

```diff
@@ -21,23 +66,143 @@ var startupCacheEventsSent = false;
  * Set by build process. Set as a global because it
  * is also used in html_cache.js.
  */
-var HTML_COOKIE_CACHE_VERSION = '2';
+var HTML_CACHE_VERSION = '2';
 
 /**
- * Max size of cookie cache segments. Set as a global because it is also used by
- * html_cache.js.
+ * A function callback is registered if the model needs to be loaded before
+ * final view determination can be done. This should be a rare event, but can
+ * happen, see localOnModelLoaded for more details. config.js looks for this
+ * callback.
  */
-var HTML_COOKIE_CACHE_MAX_SEGMENTS = 40;
-
-// Use a global to work around issue with
-// navigator.mozHasPendingMessage only returning
-// true to the first call made to it.
-window.htmlCacheRestorePendingMessage = [];
+var startupOnModelLoaded;
 
 (function() {
-  var selfNode = document.querySelector('[data-loadsrc]'),
+  var evt,
+      handlers = {},
+      pendingMessage = null,
+      startingUp = true,
+      lastEntryTime = 0,
+      handlerQueues = {
+        notification: [],
+        activity: [],
+        'request-sync': []
+      },
+      selfNode = document.querySelector('[data-loadsrc]'),
       loader = selfNode.dataset.loader,
-      loadSrc = selfNode.dataset.loadsrc;
+      loadSrc = selfNode.dataset.loadsrc,
+      // Fragment ID parsing commented out for now until further testing can be
+      // done on non-trivial entry points, like manual_config.
+      startupData = {}; //parseFragmentId();
+
+  /**
+   * startupOnModelLoaded is set to this function if the
+   * data_has_account localStorage value is not known. Called eventually by
+   * config.js once the model module has been loaded.
+   */
+  function localOnModelLoaded(model, callback) {
+    model.once('acctsSlice', function(acctsSlice) {
```

Comment:
Would it be safer for this to be a call to latestOnce()?  If not or undesired, a comment seems appropriate.